plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcPageSourceFactory.java

```diff
@@ -364,9 +366,16 @@ else if (column.getBaseHiveColumnIndex() < fileColumns.size()) {
                 Type readType = column.getType();
                 if (orcColumn != null) {
                     int sourceIndex = fileReadColumns.size();
-                    columnAdaptations.add(ColumnAdaptation.sourceColumn(sourceIndex));
+                    Optional<TypeCoercer<?, ?>> coercer = createCoercer(orcColumn.getColumnType(), readType, getTimestampPrecision(session));
+                    if (coercer.isPresent()) {
+                        fileReadTypes.add(coercer.orElseThrow().getFromType());
+                        columnAdaptations.add(ColumnAdaptation.coercedColumn(sourceIndex, coercer.orElseThrow()));
```

Comment:
nit: `orElseThrow` -> `get`

`orElseThrow` is recommended where you don't have `isPresent` check (for `get` there is an IDE warning)
`get` is shorter. yes, i know is has api-note "The preferred alternative to this method is `orElseThrow`". Well, Optionals are "recommended for method return type" and do we care?